app/main.py

```diff
@@ -1,3 +1,78 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+import datetime
+from app.shop import Shop
+from app.customer import Customer
+from app.car import Car
+
+
+def shop_trip() -> None:
+    with open("app/config.json", "r") as config_file:
+        config_dict = json.load(config_file)
+        fuel_price = config_dict["FUEL_PRICE"]
+        customers = [
+            Customer(
+                customer["name"],
+                customer["product_cart"],
+                customer["location"],
+                customer["money"],
+                Car(customer["car"]["brand"],
+                    customer["car"]["fuel_consumption"]))
+            for customer in config_dict["customers"]
+        ]
+        shops = [
+            Shop(
+                shop["name"],
+                shop["location"],
+                shop["products"]
+            ) for shop in config_dict["shops"]
+        ]
+        for customer in customers:
+            home_location = customer.location
+            min_total_costs = 10 ** 6
```

Comment:
So bad to calculate this in cycle move to CONSTANT it
BTW, it is not good practice to set some value, min_total_costs it must be first

Reply:
done